File: app/src/main/java/workshop/akbolatss/tagsnews/model/DBOpenHelper.java
package workshop.akbolatss.tagsnews.model;

import android.content.Context;

import org.greenrobot.greendao.database.Database;

import workshop.akbolatss.tagsnews.model.dao.DaoMaster;

/**
 * Author: Akbolat Sadvakassov
 * Date: 28.01.2018
 */

public class DBOpenHelper extends DaoMaster.OpenHelper {

    public DBOpenHelper(Context context, String name) {
        super(context, name);
    }

    @Override
    public void onUpgrade(Database db, int oldVersion, int newVersion) {
        super.onUpgrade(db, oldVersion, newVersion);
        switch (oldVersion) {
            case 1:

                break;

            case 2:
                break;
        }
    }
}

File: app/src/main/java/workshop/akbolatss/tagsnews/model/DBOpenHelper.kt
package workshop.akbolatss.tagsnews.model

import android.content.Context

import org.greenrobot.greendao.database.Database

import workshop.akbolatss.tagsnews.model.dao.DaoMaster

/**
 * Data Base helper. Used when entity is changed.
 */
class DBOpenHelper(context: Context, name: String) : DaoMaster.OpenHelper(context, name) {

    override fun onUpgrade(db: Database?, oldVersion: Int, newVersion: Int) {
        super.onUpgrade(db, oldVersion, newVersion)
        when (oldVersion) {
            1 -> {
                //Do when your entity is changed
            }

            2 -> {
            }
        }
    }
}
